src/ddmd/backend/gloop.c

```diff
@@ -1005,15 +1005,34 @@ STATIC void markinvar(elem *n,vec_t rd)
                 markinvar(n->E1,rd);
                 tmp = vec_clone(rd);
                 markinvar(n->E2,tmp);
-                vec_orass(rd,tmp);              /* rd |= tmp            */
+                if (!el_noreturn(n->E2))
+                    vec_orass(rd,tmp);              // rd |= tmp
                 vec_free(tmp);
                 break;
         case OPcolon:
         case OPcolon2:
                 tmp = vec_clone(rd);
-                markinvar(n->E1,rd);
-                markinvar(n->E2,tmp);
-                vec_orass(rd,tmp);              /* rd |= tmp            */
+                switch (el_noreturn(n->E1) * 2 | el_noreturn(n->E2))
+                {
+                    case 0:
```

Comment:
Actually, after this was pulled, I was going to refactor to use `el_returns()` instead of `el_noreturns()`, as the extra nots drove me batty.